### PR TITLE
feat: Rollback version endpoint for source

### PR DESCRIPTION
**Pull request by hosekpeter**
Jira: [PSGO-779](https://keboola.atlassian.net/browse/PSGO-779)

**Changes:**
- added an endpoint
- added E2E tests

-----------

[PSGO-779]: https://keboola.atlassian.net/browse/PSGO-779?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Matovidlo — APPROVED** [review]
LGTM

File: test/stream/api/source/rollback-version-404/005-poll-rollback/expected-response.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "taskId": "api.rollback.sourceVersion/%%TEST_DEFAULT_BRANCH_ID%%/my-source/%s",
+  "type": "api.rollback.sourceVersion",
+  "url": "https://stream.keboola.local/v1/tasks/api.rollback.sourceVersion/%%TEST_DEFAULT_BRANCH_ID%%/my-source/%s",
+  "status": "error",
+  "isFinished": true,
+  "createdAt": "%s",
+  "finishedAt": "%s",
+  "duration": %d,
+  "error": "Source version \"my-source/0000012345\" not found in the branch."
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/rollback-version-404/005-poll-rollback/expected-response.json:10]: The check if version exists should happen immediately and not only in the background task. `004-rollback-version` should return `404`. Enable and Disable on Source and Sink work this way as well.

**Review by jachym-tousek-keboola — CHANGES_REQUESTED** [review]

File: test/stream/api/source/rollback-version-ok/005-rollback-version/request.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "path": "/v1/branches/default/sources/my-source/versions/1/rollback",
+  "method": "PUT",
+  "headers": {
+    "Content-Type": "application/json",
+    "X-StorageApi-Token": "%%TEST_KBC_STORAGE_API_TOKEN%%"
+  }
+}
+
+
+
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/rollback-version-ok/005-rollback-version/request.json:11]:
```suggestion

```

File: test/stream/api/source/rollback-version-ok/007-list-source-versions/expected-response.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  ],
+  "page": {
+    "limit": 100,
+    "totalCount": 3,
+    "afterId": "",
+    "lastId": "%s"
+  }
+}
+
```
Comment (jachym-tousek-keboola) [inline test/stream/api/source/rollback-version-ok/007-list-source-versions/expected-response.json:50]:
```suggestion
```

File: test/stream/api/source/rollback-version-ok/005-rollback-version/request.json (hunk 2 above)
Comment (jachym-tousek-keboola) [inline test/stream/api/source/rollback-version-ok/005-rollback-version/request.json:11]:
```suggestion
```

**Pull request merged**